Functional: Rename `ClosureRefs` to `CapturedVars`

tehrengruber opened the pull request:
This PR renames the `ClosureRefs` datastructure in the functional frontend to `CapturedVars`. The name `ClosureRefs` is not optimal (in my opinion) as the noun reference has a different meaning in computer science that is not meant here. The datastructure holds essentially the same information as `inspect.getclosurevars` which returns a named tuple `ClosureVars`. To avoid a name collision, the name `CapturedVars` was choosen (expressing that it holds the variables captured inside a function). Additionally the `closure` attribute of `past.Program` and `foast.FieldOperator` has been renamed, both because it does not actually store a closure and to reflect the change.

egparedes commented on src/functional/ffront/foast_passes/type_deduction.py:
```diff
@@ -78,15 +78,15 @@ class FieldOperatorTypeDeduction(NodeTranslator):
 
     >>> sdef = SourceDefinition.from_function(example)
-    >>> cref = ClosureRefs.from_function(example)
+    >>> cref = CapturedVars.from_function(example)
```
Make sure to handle a potential conflict with the `cref` name, since it has been also renamed in other PR.
```suggestion
    >>> c_vars = CapturedVars.from_function(example)
```

egparedes reviewed (commented):
Please add a description to the PR and fix a missing variable renaming in the docstring examples.

tehrengruber commented:
I tried to keep the overhead low and hence didn't write a description. I have added one now so that everyone can follow the motive. Only change since last review is that I renamed the `closure` attribute in both `past.Program` and `foast.FieldOperator`.

egparedes reviewed (approved):
It looks good to me.
